portal-impl/src/com/liferay/portal/xml/NodeImpl.java

```diff
@@ -311,6 +313,22 @@ public void write(Writer writer) throws IOException {
 		_node.write(writer);
 	}
 
+	public void write(OutputStream outputStream, String indent, boolean expandEmptyElements,
+			boolean trimText) throws IOException {
+
+		OutputFormat outputFormat = OutputFormat.createPrettyPrint();
+
+		outputFormat.setExpandEmptyElements(expandEmptyElements);
+		outputFormat.setIndent(indent);
+		outputFormat.setLineSeparator(StringPool.NEW_LINE);
+		outputFormat.setTrimText(trimText);
+
+		XMLWriter xmlWriter = new XMLWriter(
+			outputStream, outputFormat);
+
+		xmlWriter.write(_node);
+	}
+
```

Comment:
@tinatian
Since the new XMLHelper interface cannot usage dom4j.Node, I needed a way to access the wrapped _node inside NodeImpl so that I can do this custom formatting. 
Not sure if this is ok to do since I made a public api